roles/ssh-manager/tasks/main.yml

```
@@ -0,0 +1,47 @@
+#SPDX-License-Identifier: MIT-0
+---
+- name: Get passwd database on target to find user home directories
+  ansible.builtin.getent:
+    database: passwd
+  register: getent_passwd_result
+  delegate_to: "{{ target_host | default(omit) }}"
+
+- name: Find applicable SSH roles for current target host
+  ansible.builtin.set_fact:
+    applicable_roles: >-
+      {{ ssh_roles
+         | selectattr('hosts', 'contains', target_host | default(inventory_hostname))
+         | list }}
+  delegate_to: "{{ target_host | default(omit) }}"
```

This is where the lookup can be used.

```suggestion
- name: Find applicable SSH roles
  ansible.builtin.set_fact:
    applicable_roles: >-
      {{ ssh_roles
         | <accept roles for which inventory_hostname is contained in query('inventory_hostnames', role[role_name]['hosts'])> }}"
         | list }}
```